design/design.md

```diff
@@ -0,0 +1,35 @@
+## Design Decisions
```

Comment:
Provide more background. Imagine that this is the design document you are providing to colleagues: what is this project, who worked on it, etc.